conda_smithy/cli.py

```diff
@@ -123,10 +131,21 @@ def __call__(self, args):
             )
 
         # Get some information about the source recipe.
-        if args.recipe_directory:
+        # detect if it's old recipe or new one
+        meta: Union[MetaData, RattlerMetaData]
+        try:
```

Comment:
Don't use a try statement. I would suggest one of the two.
1. Check that there's a `<recipe_dir>/recipe.yaml`
2. Check for `conda_build_tool` in `<recipe_dir>/conda-forge.yml`

Reply:
thanks, will change it

Reply:
Implemented this using has_rattler_recipe
https://github.com/conda-forge/conda-smithy/pull/1876/files#diff-af50d500653645ceb4a41789ee02aec35fa4c46b48c4635bb611ed88a5d3c3b9R141

let me know if it's ok